models/ROMS_rutgers/model_mod.f90

```diff
@@ -0,0 +1,1716 @@
+call nc_begin_define_mode(ncid)
+call nc_add_global_creation_time(ncid)
+call nc_add_global_attribute(ncid, "model_source", source)
+call nc_add_global_attribute(ncid, "model", "ROMS")
+call nc_end_define_mode(ncid)
+
+! Flush the buffer and leave netCDF file open
+call nc_synchronize_file(ncid)
+
+end subroutine nc_write_model_atts
+
+
+!-----------------------------------------------------------------------
+! Using a pre-defined factor, create ensemble perturbations around 
+! a single state.  
+
+subroutine pert_model_copies(state_ens_handle, ens_size, pert_amp, interf_provided)
+
+type(ensemble_type), intent(inout) :: state_ens_handle
+integer,             intent(in)    :: ens_size
+real(r8),            intent(in)    :: pert_amp
+logical,             intent(out)   :: interf_provided
+
+integer             :: var_type
+integer             :: j,i 
+integer(i8)         :: dart_index
+type(location_type) :: location
+
+! Storage for a random sequence for perturbing a single initial state
+type(random_seq_type) :: random_seq
+
+if (.not. module_initialized) call static_init_model
+
+interf_provided = .true.
+
+! Initialize random number sequence
+call init_random_seq(random_seq, my_task_id())
+
+! Perturb the salinity and temperature fields
+do i = 1, state_ens_handle%my_num_vars
+   dart_index = state_ens_handle%my_vars(i)
+   call get_state_meta_data(dart_index, location, var_type)
+
+   if (var_type == QTY_TEMPERATURE .or. var_type == QTY_SALINITY) then
+      do j = 1, ens_size
+         state_ens_handle%copies(j,i) = random_gaussian(random_seq, &
+            state_ens_handle%copies(j,i), perturbation_amplitude)
+      enddo
+   endif
+
+enddo
+
+end subroutine pert_model_copies
+
+
+!-----------------------------------------------------------------------
+! Writes the time of the current state and (optionally) the time
+! to be conveyed to ROMS to dictate the length of the forecast.
+! This file is then used by scripts to modify the ROMS run.
+! The format in the time information is totally at your discretion.
+!
+! ncid:        File id
+! model_time:  The current time of the model state
+! adv_to_time: The time in the future of the next assimilation.
+
+subroutine write_model_time(ncid, model_time, adv_to_time)
+
+character(len=*), parameter :: routine = 'write_model_time'
+
+integer,         intent(in)           :: ncid
+type(time_type), intent(in)           :: model_time
+type(time_type), intent(in), optional :: adv_to_time
+
+integer         :: io, varid, seconds, days
+type(time_type) :: origin_time, deltatime
+real(digits12)  :: run_duration
+
+if (.not. module_initialized) call static_init_model
+
+if (present(adv_to_time)) then
+   string3 = time_to_string(adv_to_time)
+   
+   write(string1, *) 'ROMS/DART not configured to advance ROMS.'
+   write(string2, *) 'called with optional advance_to_time of'
+   
+   call error_handler(E_ERR, routine, string1, &
+              source, text2=string2,text3=string3)
+endif
```

Comment:
Why have the adv_to_time optional argument at all?

I think this is a relic of when dart used to write its own restart files then run model_to_dart. 
Nothing calls write_model_time with adv_to_time so you can remove it from this model_mod.